.github/workflows/pytest-pyspark.yml

```diff
@@ -35,3 +35,49 @@ jobs:
         run: uv pip freeze
       - name: Run pytest
         run: pytest tests --cov=narwhals/_spark_like --cov-fail-under=95 --runslow --constructors pyspark
+
+
+  pytest-pyspark-connect-constructor:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'pyspark-connect') || contains(github.event.pull_request.labels.*.name, 'release') }}
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11"]
+        os: [ubuntu-latest]
+    env:
+      SPARK_VERSION: 3.5.5
+      SPARK_PORT: 15002
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: "true"
+          cache-suffix: ${{ matrix.python-version }}
+          cache-dependency-glob: "pyproject.toml"
+      - name: install-reqs
+        run: uv pip install -e . --group core-tests --group extra --system
+      - name: install pyspark
+        run: echo "setuptools<78" | uv pip install -b - -e . "pyspark[connect]==3.5.5" --system
+      - name: show-deps
+        run: uv pip freeze
+
+      - name: Download Spark
+        run: |
+          curl -sL https://downloads.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.tgz
+          tar xzf spark-${SPARK_VERSION}-bin-hadoop3.tgz
+          echo "SPARK_HOME=$PWD/spark-${SPARK_VERSION}-bin-hadoop3" >> $GITHUB_ENV
+          echo "$PWD/spark-${SPARK_VERSION}-bin-hadoop3/bin" >> $GITHUB_PATH
```

Comment:
So apparently this is not a good idea.

Searching for `start-connect-server.sh` in across all public github actions (yaml files in `.github` folder) here is what we get: [search](https://github.com/search?q=%22start-connect-server.sh%22+language%3AYAML+path%3A%2F%5E%5C.github%5C%2Fworkflows%5C%2F%2F&type=code&p=1).

Which to me does not really look great since all the repos are `apache/spark` forks 🥲 with [one exception](https://github.com/golfalot/sedona/blob/974fc0748c747a6692751de754bf3cd90af30744/.github/workflows/python.yml#L139), which however uses that string to skip the spark connect test.

I will sleep on this and keep looking for alternatives 🤔

Reply:
Exciting news! It seems to work with the latest commit (https://github.com/narwhals-dev/narwhals/pull/2417/commits/05e0bcc4c1545e2a8fef787a6900e145e2340a74). It just takes a while to set up